src/sqlfluff/utils/reflow/README.md

```diff
@@ -0,0 +1,219 @@
+# Reflow Primer
```

Comment:
I don't see any mention of templating in this file. Will computing and applying reflow fixes work similarly to the current rules and core linting engine? I.e.:
* Some rules have additional logic to behave intelligently in or near templated code.
* The core linting engine prevents changing templated code or copying code from a templated region to literal code, or to other templated regions.

Reply:
Kind of yes.

The core linting engine code around not changing templated code still applies :+1:.

However, I do think that with the new markers added in #3766 and the pre-existing `TemplatedSegment`, that we can remove any template specific code in rules. I might be wrong, but this is my hope.